### PR TITLE
Docs 594 additional infowindow arguments

### PR DESCRIPTION
**Pull request by csobier**
I'm working on this as part of [Docs issue#594](https://github.com/CartoDB/docs/issues/594)- WIP...

### CODE REVIEW

File: doc/api_methods.md [hunk 1]
```diff
@@ -70,14 +70,14 @@ Returns an array of layers in the map. The first is the base layer.
 
 ### vis.addOverlay(_options_)
 
-Adds an overlay to the map that can be either a zoom control, a tooltip or an infobox.
+Adds an overlay to the map that can be either a zoom control, a tooltip or an infowindow (hover action).
```
Comment (csobier) [inline doc/api_methods.md]: @rochoa , can you confirm if "infobox" is the same as "infowindow"? I changed for consistency....

File: doc/api_methods.md [hunk 2]
```diff
@@ -114,20 +129,62 @@ An overlay is internally a [Backbone.View](http://backbonejs.org/#View) so if yo
...
+--- | ---
+&#124;_ type | defines the `vis.addOverlay` [option](#visaddoverlayoptions) as infowindow.
+&#124;_ layer | cartodb layer (or sublayer).
+&#124;_ template | the tooltip content wrapper for the hover template.
+&#124;_ position | defines the position of the hover action.
+&#124;_ fields | array of column names.
+
+**REVIEWER**: WHAT ARE REQUIRED OPTIONS? ALSO, WHAT ARE SOME OF THE OTHER POSSIBLE VALUES?
+
+**Tip:** See [How can I use CartoDB.js to create and style infowindows?](/faqs/infowindows/#how-can-i-use-cartodb.js-to-create-and-style-infowindows) for an overview of how to create infowindows.
```
Comment (csobier) [inline doc/api_methods.md]: @rochoa , can you review all these additions? And/or give me guidance if any of the above line items need clarification? 
I've scheduled these updates to be released as part of the next Sprint milestone so your feedback is imperative.

**Comment by rochoa** [comment]
@alonsogarciapablo will help you with this one.

**Comment by csobier** [comment]
@alonsogarciapablo , I need review of this branch please?

File: doc/api_methods.md (hunk 1 above)
Comment (alonsogarciapablo) [inline doc/api_methods.md]: `infofox` and `infowindow` are not the same thing. There's an example of an inbox in [this example](https://github.com/CartoDB/cartodb.js/tree/develop/examples/TheHobbitLocations) (I hope it still works).

File: doc/api_methods.md [hunk 3]
```diff
@@ -85,6 +85,22 @@ If no layer is provided, the overlay will be added to the first layer of the vis
 
 An overlay object, see [vis.Overlays](#visoverlays).
 
+#### Example (Infowindow with Hover Action)
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: This is what we usually referred to as a tooltip.

File: doc/api_methods.md [hunk 4]
```diff
@@ -85,6 +85,22 @@ If no layer is provided, the overlay will be added to the first layer of the vis
...
+
+The following example displays how to enable infowindow interactivity with the mouse "hover" action.
+
+{% highlight html %}
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: I'd personally indent this code like:
```
{% highlight html %}
layer.leafletMap.viz.addOverlay({
  type: 'tooltip',
  layer: sublayer,
  template: '<div class="cartodb-tooltip-content-wrapper"><img style="width: 100%" src={{_url}}>{{name}}, {{age}}, {{city}}, {{country}}</div>', 
  position: 'bottom|right',
  fields: [{ name: 'name' }]
});
{% endhighlight %}
```

File: doc/api_methods.md [hunk 5]
```diff
@@ -70,14 +70,14 @@ Returns an array of layers in the map. The first is the base layer.
...
-Adds an overlay to the map that can be either a zoom control, a tooltip or an infobox.
+Adds an overlay to the map that can be either a zoom control, a tooltip or an infowindow (hover action).
 
 #### Arguments
 
 Option | Description
 --- | ---
 layer | layer from the visualization where the overlay should be applied (optional)
-type | zoom / tooltip / infobox
+type | zoom / tooltip / [infowindow with hover action](#visaddinfowindowmap-layer-fields--options)
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: These are actually the string literals that you can set as types. `infobox` is actually one of them, so I'd live it the way it was.

File: doc/api_methods.md [hunk 6]
```diff
@@ -114,20 +130,62 @@ An overlay is internally a [Backbone.View](http://backbonejs.org/#View) so if yo
...
+**Note:** By default, the `vis.addInfowindow` code displays interactivity for the infowindow with the "click" action. If you are using the "hover" action, you will still have to define the infowindow options with `viz.addInfowindow`, and specify additional parameters for the hover action with the `vis.Overlays` code.
 
 #### Arguments
 
 Option | Description
 --- | ---
-map | native map object or leaflet
-layer | cartodb layer (or sublayer)
-fields | array of column names
+map | native map object or leaflet.
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: This method accepts 4 arguments: `map`, `layer`, `fields`, and `options`. `options` is the only optional argument. 

The only options I'd document are:
-  `options.infowindowTemplate` which allows users to set the HTML of the template. 
- `options.templateType` which sets the type of template (`underscore` or `mustache`).
- `options.triggerEvent` which can be one of `click` or `hover`.

File: doc/api_methods.md [hunk 7]
```diff
@@ -114,20 +130,62 @@ An overlay is internally a [Backbone.View](http://backbonejs.org/#View) so if yo
...
 #### Returns
 
 An infowindow object, see [sublayer.infowindow](#sublayerinfowindow)
 
+#### Example
+
+The following example displays how to enable infowindow interactivity with the "click" action.
+
+{% highlight html %}
+ cartodb.vis.Vis.addInfowindow(map, sublayer, ['cartodb_id', 'lat', 'lon', 'name'],{
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: I'd change the indentation of this example as suggested before.

File: doc/api_methods.md [hunk 8]
```diff
@@ -114,20 +130,62 @@ An overlay is internally a [Backbone.View](http://backbonejs.org/#View) so if yo
...
+           infowindowTemplate: $('#infowindow_template').html(),
+           templateType: 'mustache'
+         });
+{% endhighlight %}
+
+**Tip:** See [`vis.addOverlay`](#visaddoverlayoptions) for how to define a hover action for an infowindow.
+
+#### Example (Infowindow with Hover Action)
```
Comment (alonsogarciapablo) [inline doc/api_methods.md]: IMO, this example doesn't make too much sense for me since this is the section about `cartodb.vis.Vis.addInfowindow`.

**Comment by alonsogarciapablo** [comment]
@csobier I left some comments. Thanks!

File: doc/api_methods.md (hunk 1 above)
Comment (csobier) [inline doc/api_methods.md]: I cannot view the example. 
What is the difference between infobox and infowindow exactly? Are they not related at all?

Comment (alonsogarciapablo) [inline doc/api_methods.md]: Here's how an infobox looks like:

screen shot 2016-02-01 at 16 53 20

It basically appears when you hover a feature and displays some info about that featured, but it's always displayed in the same position (bottom left of the window in :point_up: case).

Comment (csobier) [inline doc/api_methods.md]: Ah, nice!  So infobox is similar to infowindow (hover action), but without any positioning options.  Thanks! I will clarify all this in the infowindow doc.

**Comment by csobier** [comment]
Thanks @alonsogarciapablo! I applied some minor edits based on your feedback.

**Comment by csobier** [comment]
Okay @alonsogarciapablo , I see your point. Instead of re-listing the methods for each trigger event, I just added a more ambiguous hyperlink for hover.

**Comment by csobier** [comment]
@alonsogarciapablo , I've reverted some changes since @javisantana confirmed that `cdb.vis.Vis.addInfowindow` and "Hover infowindow can only be done with `.addOverlay`. triggerEvent is no longer mentioned. Let me know is okay?

**Comment by csobier** [comment]
@alonsogarciapablo , can you tell what section is wrong? Maybe I don't understand?

**Comment by alonsogarciapablo** [comment]
[This whole section](http://docs.cartodb.com/cartodb-platform/cartodb-js/api-methods/#visaddinfowindowmap-layer-fields--options)

**Comment by csobier** [comment]
@alonsogarciapablo , I think this should do it!  I've moved `cartodb.vis.Vis.addInfowindow()` to the Specific UI Functions section. I think this makes sense as now as all the options are explained in one section.

**Comment by alonsogarciapablo** [comment]
:+1:

**Pull request merged**